tests/kafkatest/tests/streams/streams_broker_compatibility_test.py

```diff
@@ -63,13 +63,12 @@ def test_fail_fast_on_incompatible_brokers_if_eos_enabled(self, broker_version):
         self.kafka.start()
 
         processor = StreamsBrokerCompatibilityService(self.test_context, self.kafka, True)
-        processor.start()
 
-        processor.node.account.ssh(processor.start_cmd(processor.node))
         with processor.node.account.monitor_log(processor.STDERR_FILE) as monitor:
-            monitor.wait_until("Exception in thread \"main\" org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support LIST_OFFSETS ",
+            processor.start()
```

Comment:
Ditto here.

Reply:
Yes. This is a race condition -- we always need to get the monitor first -- also unclear to me, why we start the processor twice in the old code...